core/types/subtyping.cc

```diff
@@ -470,6 +454,31 @@ TypePtr Types::lub(const GlobalState &gs, const TypePtr &t1, const TypePtr &t2)
     return lubGround(gs, t1, t2);
 }
 
+TypePtr lubProxyNonProxy(const GlobalState &gs, const TypePtr &t1, const TypePtr &t2) {
```

Comment:
Is there a way to either statically type or (in debug build) assert t1 is proxy and t2 is non-proxy?